Mathlib/Analysis/Meromorphic/TrailingCoefficient.lean

```diff
@@ -0,0 +1,258 @@
+  apply Filter.EventuallyEq.eq_of_nhds
+  rw [← h'₁.continuousAt.eventuallyEq_nhds_iff_eventuallyEq_nhdsNE h₂.continuousAt]
+  filter_upwards [h₄, h'₃, self_mem_nhdsWithin] with y h₁y h₂y h₃y
+  rw [← sub_eq_zero]
+  rwa [h₂y, ← sub_eq_zero, ← smul_sub, smul_eq_zero_iff_right] at h₁y
+  simp_all [zpow_ne_zero, sub_ne_zero]
+
+/--
+Variant of `meromorphicTrailingCoeffAt_of_order_eq_finite`: Definition of the trailing coefficient
+in case where `f` is meromorphic of finite order and a presentation is given.
+-/
+@[simp]
+lemma AnalyticAt.meromorphicTrailingCoeffAt_of_order_eq_finite₁ (h₁ : AnalyticAt 𝕜 g x)
+    (h₂ : g x ≠ 0) (h₃ : f =ᶠ[𝓝[≠] x] fun z ↦ (z - x) ^ n • g z) :
+    meromorphicTrailingCoeffAt f x = g x := by
+  have h₄ : MeromorphicAt f x := by
+    rw [MeromorphicAt.meromorphicAt_congr h₃]
+    fun_prop
+  have : meromorphicOrderAt f x = n := by
+    simp only [meromorphicOrderAt_eq_int_iff h₄, ne_eq, zpow_natCast]
+    use g, h₁, h₂
+    exact h₃
+  simp_all [h₄.meromorphicTrailingCoeffAt_of_order_eq_finite h₁, this]
+
+/--
+If `f` is analytic and does not vanish at `x`, then the trailing coefficient of `f` at `x` is `f x`.
+-/
+@[simp]
+lemma AnalyticAt.meromorphicTrailingCoeffAt_of_nonvanish (h₁ : AnalyticAt 𝕜 f x) (h₂ : f x ≠ 0) :
+    meromorphicTrailingCoeffAt f x = f x := by
+  rw [h₁.meromorphicTrailingCoeffAt_of_order_eq_finite₁ (n := 0) h₂]
+  filter_upwards
+  simp
+
+/--
+If `f` is meromorphic at `x`, then the trailing coefficient of `f` at `x` is the limit of the
+function `(· - x) ^ (-h₁.order.untop₀) • f`.
+-/
+lemma MeromorphicAt.meromorphicTrailingCoeffAt_eq_limit (h : MeromorphicAt f x) :
+    Tendsto ((· - x) ^ (-(meromorphicOrderAt f x).untop₀) • f) (𝓝[≠] x)
+      (𝓝 (meromorphicTrailingCoeffAt f x)) := by
+  by_cases h₂ : meromorphicOrderAt f x = ⊤
+  · simp_all only [WithTop.untop₀_top, neg_zero, zpow_zero, one_smul,
+      meromorphicTrailingCoeffAt_of_order_eq_top]
+    apply Tendsto.congr' (f₁ := 0)
+    · filter_upwards [meromorphicOrderAt_eq_top_iff.1 h₂] with y hy
+      simp_all
+    · apply Tendsto.congr' (f₁ := 0) (by rfl) continuousWithinAt_const.tendsto
+  obtain ⟨g, h₁g, h₂g, h₃g⟩ := (meromorphicOrderAt_ne_top_iff h).1 h₂
+  apply Tendsto.congr' (f₁ := g)
+  · filter_upwards [h₃g, self_mem_nhdsWithin] with y h₁y h₂y
+    rw [zpow_neg, Pi.smul_apply', Pi.inv_apply, Pi.pow_apply, h₁y, ← smul_assoc, smul_eq_mul,
+      ← zpow_neg, ← zpow_add', neg_add_cancel, zpow_zero, one_smul]
+    left
+    simp_all [sub_ne_zero]
+  · rw [meromorphicTrailingCoeffAt_of_order_eq_finite h h₁g h₂ h₃g]
+    apply h₁g.continuousAt.continuousWithinAt
+
+/-!
+## Elementary Properties
+-/
+
+/--
+If `f` is meromorphic of finite order at `x`, then the trailing coefficient is not zero.
+-/
+lemma MeromorphicAt.zero_ne_meromorphicTrailingCoeffAt (h₁ : MeromorphicAt f x)
+    (h₂ : meromorphicOrderAt f x ≠ ⊤) :
+    0 ≠ meromorphicTrailingCoeffAt f x := by
```

Comment:
I'd rather have the more natural
```suggestion
    meromorphicTrailingCoeffAt f x ≠ 0 := by
```
(and the lemma name should be fixed accordingly).

Reply:
Implemented.